wger.py

```diff
@@ -127,186 +161,188 @@ def create_exercise(self):
         category_id = str([i['id'] for i in categories if i['name'] == category][0])
         equipment_id = str([i['id'] for i in equipments if i['name'] == equipment][0])
         muscle_id =str([i['id'] for i in muscles if i['name'] == muscle][0])
-        data={'name':name, 'description':description, 'category':category_id, 'equipment': equipment_id, 'muscles':muscle_id,
+        data={'name':name,
+              'description':description,
+              'category':category_id,
+              'equipment': equipment_id,
+              'muscles':muscle_id,
               'license_author': author}
         response=self.post_info('exercise/',data)
         return response
 
-    def select_exercises(self,workout,day):
+    def select_exercises(self,workout,day,total_ex,ex_musclegroup):
         """
         The function selects 6 exercises for a given training day.
         There needs to be no more than 2 exercises per muscle group per day.
         Also, there needs to be at least 1 exercice for back and one for front per day.
         """
-        id_workout=""
-        id_day=""
+        workouts=self.get_info('workout/').get('results')
+        days=self.get_info('day/').get('results')
         selected = []
 
-        for i in self.get_info('workout/')['results']:
-            if i['name']==workout:
-                id_workout=i['id']
-        if id_workout == "":
-            return "Workout not found"
+        id_workout=self.match(workouts, workout, 'name', 'id')
+        if id_workout=="":
+            return False
 
-        for i in self.get_info('day/')['results']:
-            if i['description'] == day:
-                id_day = i['id']
-        if id_day == "":
-            return "Training not found"
+        id_day = self.match(days,day, 'description', 'id')
+        if id_workout == "":
+            return False
 
-        categories =dict( [(i['name'],0) for i in self.get_info('exercisecategory/')['results']])
-        exercises=[(i['id'],i['category']['name']) for i in self.get_info("exerciseinfo/?limit=100&offset=100",)['results']]
+        categories =dict( [(i['name'],0) for i in self.get_info('exercisecategory/').get('results')])
+        exercises=[(i['id'],i['category']['name']) for i in self.get_info("exerciseinfo/",100).get('results')]
         selected.append(random.choice([i for i in exercises if i[1]==('Chest')]))
         categories['Chest'] += 1
         selected.append(random.choice([i for i in exercises if i[1] == ('Back')]))
```

Comment:
Lines 193-194 and 195-196 are duplicates of one another. You can eliminate the duplicate code with a loop.